Port the script to Imgur API version 3. Also use HTTPS where appropriate.

diff -ru a/imgurbash.sh b/imgurbash.sh
--- a/imgurbash.sh	2016-04-08 17:23:27.231752818 +0300
+++ b/imgurbash.sh	2016-04-08 18:12:27.073429100 +0300
@@ -25,8 +25,8 @@
 # stderr). If you have xsel or xclip the URLs will also be put on the X 
 # selection, which you can usually paste with a middle click.
 
-# API Key provided by Alan@imgur.com
-apikey="b3625162d3418ac51a9ee805b1840452"
+# This client id is for imgurbash on Arch Linux ONLY
+client_id="7275da65e609b81"
 
 # function to output usage instructions
 function usage {
@@ -37,12 +37,6 @@
 	echo "easy pasting." >&2
 }
 
-# check API key has been entered
-if [ "$apikey" = "Your API key" ]; then
-	echo "You first need to edit the script and put your API key in the variable near the top." >&2
-	exit 15
-fi
-
 # check arguments
 if [ "$1" = "-h" -o "$1" = "--help" ]; then
 	usage
@@ -75,26 +69,26 @@
 	fi
 
 	# upload the image
-	response=$(curl -F "key=$apikey" -H "Expect: " -F "image=@$file" \
-		http://imgur.com/api/upload.xml 2>/dev/null)
+	response=$(curl -H "Authorization: Client-ID $client_id" -H "Expect: " -F "image=@$file" \
+		https://api.imgur.com/3/upload.xml 2>/dev/null)
 	# the "Expect: " header is to get around a problem when using this through 
 	# the Squid proxy. Not sure if it's a Squid bug or what.
 	if [ $? -ne 0 ]; then
 		echo "Upload failed" >&2
 		errors=true
 		continue
-	elif [ $(echo $response | grep -c "<error_msg>") -gt 0 ]; then
+	elif [ $(echo $response | grep -c 'success="0"') -gt 0 ]; then
 		echo "Error message from imgur:" >&2
-		echo $response | sed -r 's/.*<error_msg>(.*)<\/error_msg>.*/\1/' >&2
+		echo $response | sed -r 's/.*<error>(.*)<\/error>.*/\1/' >&2
 		errors=true
 		continue
 	fi
 
 	# parse the response and output our stuff
-	url=$(echo $response | sed -r 's/.*<original_image>(.*)<\/original_image>.*/\1/')
-	deleteurl=$(echo $response | sed -r 's/.*<delete_page>(.*)<\/delete_page>.*/\1/')
+	url=$(echo $response | sed -r 's/.*<link>(.*)<\/link>.*/\1/;s/^http:/https:/')
+	deletehash=$(echo $response | sed -r 's/.*<deletehash>(.*)<\/deletehash>.*/\1/')
 	echo $url
-	echo "Delete page: $deleteurl" >&2
+	echo "Delete page: https://imgur.com/delete/$deletehash" >&2
 
 	# append the URL to a string so we can put them all on the clipboard later
 	clip="$clip$url
